sdcm/nemesis.py

```diff
@@ -2081,31 +2089,60 @@ def modify_table_bloom_filter_fp_chance(self):  # pylint: disable=invalid-name
         """
         self._modify_table_property(name="bloom_filter_fp_chance", val=random.random() / 2)
 
-    def toggle_table_gc_mode(self):
+    def _get_user_keyspace_table(self):
         """
-            Alters a non-system table tombstone_gc_mode option.
-            Choose the alternate option of 'repair' / 'timeout'
-             (*) The other available values of 'disabled' / 'immediate' are not tested by
-             this nemesis since not applicable to a longevity test.
+            Return a random user keyspace, table, and an indication for a Materialized-view.
         """
         all_ks_cfs = self.cluster.get_non_system_ks_cf_list(db_node=self.target_node)
         non_mview_ks_cfs = self.cluster.get_non_system_ks_cf_list(db_node=self.target_node, filter_out_mv=True)
 
         if not all_ks_cfs:
             raise UnsupportedNemesis(
-                'Non-system keyspace and table are not found. toggle_table_gc_mode nemesis can\'t run')
+                'Non-system keyspace and table are not found. The nemesis can\'t run')
 
         mview_ks_cfs = list(set(all_ks_cfs) - set(non_mview_ks_cfs))
         keyspace_table = random.choice(all_ks_cfs)
         keyspace, table = keyspace_table.split('.')
+        is_mview = keyspace_table in mview_ks_cfs
+        return keyspace, table, is_mview
+
+    def modify_table_gc_mode(self):
+        """
+            Alters a non-system table tombstone_gc_mode option.
+            Choose any option out of 'repair' / 'timeout' / 'disabled' / 'immediate'
+             (*) A GC mode legend is:
+             repair = delete only tombstones created before a repair.
+             timeout = the old original mode set by gc grace period parameter.
+             immediate = delete tombstone immediately on compaction.
+             disabled = tombstones are not deleted at all.
+        """
+        keyspace, table, is_mview = self._get_user_keyspace_table()
+        current_gc_mode = get_gc_mode(node=self.target_node, keyspace=keyspace, table=table)
+        new_gc_mode = random.choice([gc_mode for gc_mode in list(
+            GcMode) if gc_mode != current_gc_mode])
+        new_gc_mode_as_dict = {'mode': new_gc_mode.value}
+
+        alter_command_prefix = 'ALTER MATERIALIZED VIEW ' if is_mview else 'ALTER TABLE '
+        cmd = f"{alter_command_prefix} {keyspace}.{table} WITH tombstone_gc = {new_gc_mode_as_dict};"
+        self.log.info("Alter GC mode query to execute: %s", cmd)
+        self.target_node.run_cqlsh(cmd)
+
+    def toggle_table_standard_gc_mode(self):
```

Comment:
this method, and `modify_table_gc_mode()` are doing the same, just that `modify_table_gc_mode()` has more options as target to change, other than this one, that is just `repair` or `timeout`, right?
why do we need these 2 as different methods?

Reply:
Fixed it to a single method with a boolean parameter, hope it's not to conditional now..